horqrux/utils_noise.py

```diff
@@ -0,0 +1,257 @@
+    return kraus
+
+
+def PhaseFlip(error_probability: float) -> tuple[Array, ...]:
+    """
+    Initialize the PhaseFlip gate
+
+    The phase flip channel is defined as:
+
+    .. math::
+        \\rho \\Rightarrow (1-p) \\rho + p Z \\rho Z^{\\dagger}
+
+    Args:
+        error_probability (float): The probability of phase flip error.
+
+    Raises:
+        ValueError: If the error_probability value is not a float.
+
+    Returns:
+        tuple[Array, ...]: Kraus operators for this protocol.
+    """
+
+    if (error_probability > 1.0) or (error_probability < 0.0):
+        raise ValueError("The error_probability value is not a correct probability")
+    K0: Array = jnp.sqrt(1.0 - error_probability) * OPERATIONS_DICT["I"]
+    K1: Array = jnp.sqrt(error_probability) * OPERATIONS_DICT["Z"]
+    kraus: tuple[Array, ...] = (K0, K1)
+    return kraus
+
+
+def Depolarizing(error_probability: float) -> tuple[Array, ...]:
+    """
+    Initialize the Depolarizing gate.
+
+    The depolarizing channel is defined as:
+
+    .. math::
+        \\rho \\Rightarrow (1-p) \\rho
+            + p/3 X \\rho X^{\\dagger}
+            + p/3 Y \\rho Y^{\\dagger}
+            + p/3 Z \\rho Z^{\\dagger}
+
+    Args:
+        error_probability (float): The probability of phase flip error.
+
+    Raises:
+        ValueError: If the error_probability value is not a float.
+
+    Returns:
+        tuple[Array, ...]: Kraus operators for this protocol.
+    """
+
+    if (error_probability > 1.0) or (error_probability < 0.0):
+        raise ValueError(f"The 'error_probability' value is incorrect. Got {error_probability}.")
+    K0: Array = jnp.sqrt(1.0 - error_probability) * OPERATIONS_DICT["I"]
+    K1: Array = jnp.sqrt(error_probability / 3.0) * OPERATIONS_DICT["X"]
+    K2: Array = jnp.sqrt(error_probability / 3.0) * OPERATIONS_DICT["Y"]
+    K3: Array = jnp.sqrt(error_probability / 3.0) * OPERATIONS_DICT["Z"]
+    kraus: tuple[Array, ...] = (K0, K1, K2, K3)
+    return kraus
+
+
+def PauliChannel(error_probability: tuple[float, ...]) -> tuple[Array, ...]:
+    """
+    Initialize the PauliChannel gate.
+
+    The pauli channel is defined as:
+
+    .. math::
+        \\rho \\Rightarrow (1-px-py-pz) \\rho
+            + px X \\rho X^{\\dagger}
+            + py Y \\rho Y^{\\dagger}
+            + pz Z \\rho Z^{\\dagger}
+
+    Args:
+        error_probability (tuple[float, ...] | float): tuple containing probabilities
+            of X, Y, and Z errors.
+
+    Raises:
+        ValueError: If the probabilities values do not sum up to 1.
+
+    Returns:
+        tuple[Array, ...]: Kraus operators for this protocol.
+    """
+
+    sum_prob = sum(error_probability)
+    if sum_prob > 1.0:
+        raise ValueError("The sum of probabilities can't be greater than 1.0")
+    if any([probability > 1.0 or probability < 0.0 for probability in error_probability]):
+        raise ValueError(f"The 'error_probability' values are incorrect. Got {error_probability}.")
```

Comment:
```suggestion
        raise ValueError(f"The 'error_probability' values are incorrect. Got {probability}.")
```

I am not sure this is working though. Should be tested.